File: 1.动脑学院笔记2019/7.dn架构师/code/lsn13/DN_ALAN_LSN_13/app/src/main/java/com/dn_alan/myapplication/MainActivity.java
package com.dn_alan.myapplication;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

import com.dn_alan.myapplication.bean.Person;
import com.dn_alan.myapplication.bean.User;
import com.dn_alan.myapplication.db.BaseDao;
import com.dn_alan.myapplication.db.BaseDaoFactory;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


    }

    public void insert(View view) {
        BaseDao<User> baseDao = BaseDaoFactory.getOurInstance().getBaseDao(User.class);
        baseDao.insert(new User(1, "alan", "123"));

        BaseDao<Person> personBaseDao = BaseDaoFactory.getOurInstance().getBaseDao(Person.class);
        personBaseDao.insert(new Person(1, "alan_person", "123"));

        Toast.makeText(this, "执行成功！", Toast.LENGTH_SHORT).show();
    }
}

File: 7.dn架构师/code/lsn25/AndroidPluginDevelop/resourceplugin/src/main/java/com/east/resourceplugin/MainActivity.kt
package com.east.resourceplugin

import android.Manifest
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.east.resourceplugin.skin.SkinManager
import java.io.File
import java.io.FileOutputStream
import java.io.IOException

class MainActivity : BaseSkinActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun skin(view: View?) {
        // 从服务器上下载
        val copyResult = copyAssetAndWrite("red.skin") // 拷贝到本地缓存
        if(!copyResult){
            Toast.makeText(this,"没有拷贝成功,确保assets目录下有资源文件",Toast.LENGTH_SHORT).show()
            return
        }
        val SkinPath = File(cacheDir, "red.skin").absolutePath

        // 换肤
        SkinManager.getInstance().loadSkin(SkinPath)
    }

    fun skin1(view: View?) {
        // 恢复默认
        SkinManager.getInstance().restoreDefault()
    }


    fun skin2(view: View?) {
        // 跳转
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }


    fun copyAssetAndWrite(
        fileName: String?
    ): Boolean {
        try {
            val cacheDir = cacheDir
            if (!cacheDir.exists()) {
                cacheDir.mkdirs()
            }
            val outFile = File(cacheDir, fileName)
            if (!outFile.exists()) {
                val res = outFile.createNewFile()
                if (!res) {
                    return false
                }
            } else {
                if (outFile.length() > 10) { //表示已经写入一次
                    return true
                }
            }
            val `is` = assets.open(fileName!!)
            val fos = FileOutputStream(outFile)
            val buffer = ByteArray(1024)
            var byteCount: Int
            while (`is`.read(buffer).also { byteCount = it } != -1) {
                fos.write(buffer, 0, byteCount)
            }
            fos.flush()
            `is`.close()
            fos.close()
            return true
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return false
    }
}
